File: frontend/src/objects/ui/component/text-component.ts
import { z } from "zod";
import { nullishToUndefined } from "../../system/schema-utils.js";
import {
  DynamicTextProgramSchema,
  TextContentModeSchema,
} from "../../ui-customization/dynamic-text-program.js";
import { ComponentBindingSchema } from "./component-binding.js";
import { ComponentPositionSchema } from "./component-position.js";
import { ComponentStyleSchema, TextArtDesignSchema } from "./component-style.js";

export const TextComponentSchema = z.object({
  id: z.string().min(1),
  type: z.literal("text"),
  text: z.string(),
  textContentMode: nullishToUndefined(TextContentModeSchema),
  dynamicTextProgram: nullishToUndefined(DynamicTextProgramSchema),
  position: ComponentPositionSchema,
  style: nullishToUndefined(ComponentStyleSchema),
  artTextDesign: nullishToUndefined(TextArtDesignSchema),
  binding: nullishToUndefined(ComponentBindingSchema),
});
export type TextComponent = z.infer<typeof TextComponentSchema>;

File: backend/microservice/src/ui/objects/component/TextComponent.scala
package microservice.ui.objects.component

/** 文本组件：静态或绑定文本内容。 */
final case class TextComponent(
  id: String,
  text: String,
  position: ComponentPosition,
  style: Option[ComponentStyle],
  binding: Option[ComponentBinding]
) extends PageComponent {
  override val `type`: String = "text"
}
